### PR TITLE
Make Blaze persistence a provided dependency

### PR DESCRIPTION
**Pull request by maxhov**
Make Blaze persistence a provided dependency for blaze-expression-declarative and blaze-expression-declarative-persistence-jakarta

### CODE REVIEW

File: declarative/persistence/pom.xml [hunk 1]
```diff
@@ -29,6 +29,7 @@
             <groupId>${project.groupId}</groupId>
             <artifactId>blaze-expression-persistence</artifactId>
             <version>${project.version}</version>
+            <scope>provided</scope>
```
Comment (beikov) [inline declarative/persistence/pom.xml:32]: This one shouldn't be necessary

Comment (maxhov) [reply declarative/persistence/pom.xml:32]: As in, the entire dependency, or the scope?

Comment (beikov) [reply declarative/persistence/pom.xml]: The scope

Comment (maxhov) [reply declarative/persistence/pom.xml]: Fixed

**Pull request merged**